cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -1480,16 +1481,75 @@ public void testConcurrentCreateAppAfterTheFirstVersion() throws Exception {
     Assert.assertEquals(1 + numThreads, appEditNumber.get());
   }
 
+  @Test
+  public void testDeleteRunsStartedBefore() throws Exception {
+    ProgramId programId = NamespaceId.DEFAULT.app("test").workflow("test");
+
+    Instant ttlCutoff = Instant.ofEpochSecond(1631629389);
+
+    ProgramRunId shouldExpire = createCompletedRun(programId, ttlCutoff.minusSeconds(10));
+    ProgramRunId shouldNotExpire = createCompletedRun(programId, ttlCutoff.plusSeconds(10));
+
+    TransactionRunners.run(
+        transactionRunner,
+        context -> {
+          AppMetadataStore store = AppMetadataStore.create(context);
+          // Check that both items exist
+          Assert.assertNotNull(store.getRun(shouldExpire));
+          Assert.assertNotNull(store.getRun(shouldNotExpire));
+
+          // Run expiration process.
+          store.deleteRunsStartedBefore(ttlCutoff);
+
+          // Assert only the older run was deleted.
+          Assert.assertNull(store.getRun(shouldExpire));
```

Comment:
nit: it's a little cleaner to check this is another transaction, just to ensure the committed values are as expected.

Reply:
done